ApplicationLibCode/ProjectDataModel/StimPlanModel/RimPressureTable.h

```diff
@@ -0,0 +1,62 @@
+/////////////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (C) 2021-     Equinor ASA
+//
+//  ResInsight is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  ResInsight is distributed in the hope that it will be useful, but WITHOUT ANY
+//  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+//  FITNESS FOR A PARTICULAR PURPOSE.
+//
+//  See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+//  for more details.
+//
+/////////////////////////////////////////////////////////////////////////////////
+#pragma once
+
+#include "cafPdmChildArrayField.h"
+#include "cafPdmField.h"
+#include "cafPdmObject.h"
+
+#include "RiaStimPlanModelDefines.h"
+
+class RimPressureTableItem;
+
+class RimPressureTable : public caf::PdmObject
+{
+    CAF_PDM_HEADER_INIT;
+
+public:
+    RimPressureTable();
+    ~RimPressureTable() override;
+
+    caf::Signal<> changed;
+
+    std::vector<RimPressureTableItem*> items() const;
+    void                               insertItem( RimPressureTableItem* insertBefore, RimPressureTableItem* item );
+    void                               deleteItem( RimPressureTableItem* itemToDelete );
+    void                               deleteAllItems();
+
+    bool                usePressureTableForProperty( RiaDefines::CurveProperty curveProperty ) const;
+    caf::PdmField<bool> useForInitialPressure;
```

Comment:
Consider keeping these as private, and use method 2 from here
https://ceetronsolutions.github.io/resinsight-system-doc/system/include-fields-from-child-objects